### PR TITLE
Fix C assert calls for newlib

### PR DESCRIPTION
**Pull request by kubo39**

### CODE REVIEW

File: driver/main.cpp [hunk 1]
```diff
@@ -826,6 +826,8 @@ void registerPredefinedTargetVersions() {
       VersionCondition::addPredefinedGlobalIdent("DRuntime_Use_Libunwind");
     } else if (global.params.isUClibcEnvironment) {
       VersionCondition::addPredefinedGlobalIdent("CRuntime_UClibc");
+    } else if (global.params.isNewlibEnvironment) {
+      VersionCondition::addPredefinedGlobalIdent("CRuntime_Newlib");
```
Comment (kinke) [inline driver/main.cpp:830]: Is newlib a thing on Linux? It's currently only predefined when explicitly targeting bare-metal, see line ~906.

File: gen/runtime.cpp [hunk 2]
```diff
@@ -364,6 +364,7 @@ llvm::Function *getRuntimeFunction(const Loc &loc, llvm::Module &target,
 //                             const char *funcname);
 // uClibc:  void __assert(const char *assertion, const char *filename, int linenumber,
 //                        const char *function);
+// newlib:  void __assert(const char *file, int line, const char *failedexpr)
```
Comment (JohanEngelen) [inline gen/runtime.cpp:367]: nitpick: can you move this up, below Android? Then it's crystal clear that they have the same signature.

**Comment by JohanEngelen** [comment]
Tangential to this PR, I really think we should have at least one testcase for newlib (`/tests/driver/`). In that testcase, you can test simple things like whether the `version` identifier is indeed set, and whether C assert function is correct, stuff like that.

**Comment by JohanEngelen** [comment]
can you add a testcase for this? (it would also test newlib triple etc.)

**Comment by JohanEngelen** [comment]
lgtm

**Pull request merged**